Homework/homework2/task1_text.py

```diff
@@ -0,0 +1,133 @@
+import io
+import string
+from typing import List, Mapping, Tuple
+
+
+def u_decoder(line: str) -> str:
+    """Decode ascii chars."""
+    return line.encode().decode("unicode-escape")
+
+
+def decoded_line_without_punc(line: str) -> str:
+    """
+    Decode, remove punctuation from the string.
+
+    Add a long dash and angle quotes to a punctuation.
+    """
+    punctuation = string.punctuation + "\u2014" + "\u00bb" + "\u00ab"
+    map_char = str.maketrans(punctuation, " " * len(punctuation))
+    return u_decoder(line).translate(map_char).casefold()
+
+
+def _find_word_max_unique(word: str, max_word: str, max_len: int) -> Tuple[str, int]:
+    cur_len = len(set(word))
+    if max_len < cur_len:
+        max_len = cur_len
+        max_word = word
+    return max_word, max_len
+
+
+def _get_longest_diverse_words(file: io.StringIO()) -> List[str]:
+    long_words = []
+    for line in file:
+        max_word = ""
+        max_len = 0
+        words = decoded_line_without_punc(line).strip().split()
+        for word in words:
+            max_word, max_len = _find_word_max_unique(word, max_word, max_len)
+        long_words.append(max_word)
+    long_words.sort(key=lambda s: len(s), reverse=True)
+    file.close()
+    return long_words[1:11]
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    """Find 10 longest words consisting from the largest amount of unique symbols."""
+    file = open(file_path, "r")
+    return _get_longest_diverse_words(file)
+
+
+def _get_rarest_char(file: io.StringIO()) -> str:
+    char_count = {}
+    for line in file:
+        for ch in u_decoder(line).strip().lower():
+            char_count = _add_char_dict(ch, char_count)
+    return min(char_count, key=char_count.get)
+
+
+def get_rarest_char(file_path: str) -> str:
+    """Find rarest symbol for document."""
+    file = open(file_path, "r", encoding="utf-8")
+    return _get_rarest_char(file)
+
+
+def _count_punctuation_chars(file: io.StringIO()) -> int:
+    count = 0
+    punctuation = string.punctuation + "\u2014" + "\u00bb" + "\u00ab"
```

Comment:
`string.punctuation + "\u2014" + "\u00bb" + "\u00ab"`

You use it in amultiple palces, just extract it tothe variable.